src/components/molecules/AuthButton/index.tsx

```diff
@@ -0,0 +1,53 @@
+"use client";
+
+import Button from "@/components/atoms/Button";
+import { getCookie } from "@/utils/Cookie";
+import { useRouter } from "next/navigation";
+import React, { useState, useEffect } from "react";
+import AuthUserProfile from "./AuthUserProfile";
+
+function AuthButton(): JSX.Element | null {
+  const router = useRouter();
+  const [isLoad, setIsLoad] = useState(true);
+  const [isLogin, setIsLogin] = useState(false);
+
+  useEffect(() => {
+    setIsLoad(false);
+    if (getCookie("token")) setIsLogin(true);
+  });
+
+  if (isLoad) return null;
+
+  return (
+    <div className="flex items-center space-x-4">
+      {isLogin ? (
+        <AuthUserProfile setIsLogin={setIsLogin} />
```

Comment:
setState 함수를 Props로 내려주고 있는데 
저는 이런 방식보다는 해당 컴포넌트에서 선언된 상태는 그 컴포넌트에서만 처리하도록 하는 것을 선호합니다!
자식 컴포넌트가 받은 setState가 언제 호출될지, 어떤 값으로 업데이트할지 알기 어렵기 때문에
자식 컴포넌트는 핸들러를 받아서 호출하게 만들면 좋습니다

```suggestion
        <AuthUserProfile onClickLogout={() => setIsLogin(false)} />
```

Reply:
완료